### PR TITLE
Lambdas list

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- Add lambdas view in core-ui

**Related issue(s)**
https://github.com/kyma-project/console/issues/1275

### CODE REVIEW

File: core-ui/src/components/Lambdas/Lambdas.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        header: `Remove ${name}`,
+        body: `Are you sure you want to delete lambda "${name}"?`,
+        buttonConfirm: 'Delete',
+        buttonDismiss: 'Cancel',
+      })
+      .then(async () => {
+        try {
+          await deleteLambda({
+            variables: { name: name, namespace: namespace },
+          });
```
Comment (dariadomagala) [inline core-ui/src/components/Lambdas/Lambdas.js]: We should show notifications for such actions. There is a special function `useNotification` for that, you can check the usage of it in the `ModalWithForm` component. 
Same thing for the delete error.

File: core-ui/src/components/Lambdas/Lambdas.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  const actions = [
+    {
+      name: 'Delete',
+      handler: entry => {
+        handleLambdaDelete(entry.name, entry.namespace);
+      },
+    },
+  ];
+
+  const createLabels = labels => {
```
Comment (dariadomagala) [inline core-ui/src/components/Lambdas/Lambdas.js]: Maybe we could create a shared component for labels? We will definitely need that in the future.

File: core-ui/src/components/typechecking/CustomPropTypes.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import PropTypes from 'prop-types';
 
 //TODO: move this component to a shared "place"
-export const elementRef = PropTypes.shape({ current: PropTypes.element });
+export const elementRef = PropTypes.shape({ current: PropTypes.any });
```
Comment (dariadomagala) [inline core-ui/src/components/typechecking/CustomPropTypes.js]: Why this change?

File: core-ui/src/shared/components/GenericList/actionHandlers/simpleDelete.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  });
+}
+
+export default function handleDelete(
+  entityType,
+  entityId,
+  entityName,
+  deleteRequestFn,
+  callback,
+) {
```
Comment (dariadomagala) [inline core-ui/src/shared/components/GenericList/actionHandlers/simpleDelete.js]: Why are we not using it?

File: core-ui/src/components/typechecking/CustomPropTypes.js (hunk 3 above)
Comment (akucharska) [reply core-ui/src/components/typechecking/CustomPropTypes.js]: Anything can be ref and not only element

File: core-ui/src/components/CreateNamespaceForm/test/CreateNamespaceForm.test.js [hunk 5]
```diff
@@ -17,16 +18,22 @@ import {
 
 describe('CreateNamespaceForm', () => {
   it('Renders with minimal props', () => {
+    const { link } = createMockLink([]);
     const component = renderer.create(
-      <CreateNamespaceForm formElementRef={{ current: null }} />,
+      <MockedProvider link={link}>
```
[the remark this answers is not in the document]
Comment (akucharska) [reply core-ui/src/components/CreateNamespaceForm/test/CreateNamespaceForm.test.js]: Tests were failing

[the remark this answers is not in the document]
Comment (akucharska) [reply core-ui/src/components/CreateNamespaceForm/test/CreateNamespaceForm.test.js]: Tests were failing

File: core-ui/src/components/typechecking/CustomPropTypes.js (hunk 3 above)
Comment (akucharska) [reply core-ui/src/components/typechecking/CustomPropTypes.js]: Ref can be anything and not just an element

File: core-ui/src/shared/components/GenericList/actionHandlers/simpleDelete.js (hunk 4 above)
Comment (akucharska) [reply core-ui/src/shared/components/GenericList/actionHandlers/simpleDelete.js]: it uses different props

**Comment by akucharska** [comment]
> * would be great to have the list sensitive to namespace switch
This should start working after task fixing preload in core-ui

> * please use page header for the list view
> Also.. the columns for empty list are misplaced comparing to filled table. Results in "jumping columns" when filtering by name. (not a big deal for the first iteration though)
> Screen Shot 2019-10-10 at 11 43 28
> Screen Shot 2019-10-10 at 11 43 35
I think this would be great to do in following task (after creating shared components as it requires changes in Generic list)

File: core-ui/src/components/Lambdas/Lambdas.js (hunk 1 above)
Comment (akucharska) [reply core-ui/src/components/Lambdas/Lambdas.js]: ok

File: core-ui/package.json [hunk 6]
```diff
@@ -3,12 +3,15 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@kyma-project/luigi-core": "^0.6.4"
+    "@kyma-project/luigi-core": "^0.6.4",
+    "@apollo/react-hooks": "^3.1.2",
```
[the remark this answers is not in the document]
Comment (akucharska) [reply core-ui/package.json]: apparently not ;)

File: core-ui/src/index.scss [hunk 7]
```diff
@@ -38,6 +39,30 @@ body {
...
+  text-decoration: underline;
+  cursor: pointer;
+}
+
+.fd-panel.fd-panel--no-background {
+  background-color: transparent;
+}
+
+.y-fd-token{
+  &.y-fd-token{
```
Comment (parostatkiem) [inline core-ui/src/index.scss]:
```suggestion
 
```

File: core-ui/src/index.scss [hunk 8]
```diff
@@ -38,6 +39,30 @@ body {
...
+    &--no-button {
+      cursor: default;
+    }
+    &--no-button:after {
+      all: unset;
+    }
+    &--gap:not(:last-of-type) {
+      margin-right: 8px;    
+    }
+  }
```
Comment (parostatkiem) [inline core-ui/src/index.scss]:
```suggestion
  
```

File: core-ui/src/components/Lambdas/test/Lambdas.test.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    expect(component.find(Spinner)).toHaveLength(0);
+  });
+
+  it('Displays lambdas with their corresponding names in the table', async () => {
+    const { link } = createMockLink([allLambdasQuery]);
+    const component = mount(
+      <MockedProvider link={link}>
+        <Lambdas />
+      </MockedProvider>,
+    );
```
Comment (dariadomagala) [inline core-ui/src/components/Lambdas/test/Lambdas.test.js]: Do we need this link? If it's possible and easy to do I would stick to 
```<MockedProvider mocks={gqlMock}...```
as we already have in CreateNamespaceForm so we have one approach - it's just easier to maintain it in the future.

File: core-ui/src/shared/components/Labels/Labels.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+import { EMPTY_TEXT_PLACEHOLDER } from './../../constants';
+
+const Labels = labels => {
+  if (!labels || Object.keys(labels).length === 0) {
+    return <span>{EMPTY_TEXT_PLACEHOLDER}</span>;
+  }
+  const separatedLabels = [];
+  /* eslint-disable no-unused-vars */
+  for (const key in labels) {
+    separatedLabels.push(key + '=' + labels[key]);
```
Comment (dariadomagala) [inline core-ui/src/shared/components/Labels/Labels.js]:
```suggestion
    separatedLabels.push(`${key}=${labels[key]});
```

File: core-ui/src/testing/apollo.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import { MockLink, MockSubscriptionLink } from '@apollo/react-testing';
+import { isSubscriptionOperation } from '../apollo';
+import { ApolloLink } from 'apollo-link';
+
+export function createMockLink(mocks, addTypename = true) {
```
Comment (dariadomagala) [inline core-ui/src/testing/apollo.js]: As I wrote in the test - please check if it's needed.

File: core-ui/src/index.scss (hunk 7 above)
Comment (akucharska) [reply core-ui/src/index.scss]: I won't be changing is since it mess up with styles priority
